fix(ui): ListTransition and Presence return Node instead of DocumentFragment — breaks JSX type compatibility

## Problem

`ListTransition` and `Presence` components have return type `Node`, but `JSX.Element` is `HTMLElement | SVGElement | DocumentFragment`. Since `Node` is not assignable to `JSX.Element`, these components cannot be used as JSX components:

```
error TS2786: 'ListTransition' cannot be used as a JSX component.
  Its return type 'Node' is not a valid JSX element.
```

This causes `npx tsc --noEmit` to fail on a freshly scaffolded project (0.2.12).

## Root Cause

Both components create and return a `DocumentFragment`, but the return type annotation says `Node`:

```ts
// list-transition.ts
export function ListTransition<T>(props: ListTransitionProps<T>): Node {
  // ...
  return Object.assign(fragment, { dispose });
}

// presence.ts  
export function Presence(props: PresenceProps): Node {
  // ...
  return Object.assign(fragment, { dispose });
}
```

## Fix

Change return type from `Node` to `DocumentFragment` for both components. The functions actually return `DocumentFragment` — the annotation is just too wide.

## Found During

Developer walkthrough of `@vertz/create-vertz-app@0.2.12`.

--- a/packages/ui/src/component/list-transition.ts
+++ b/packages/ui/src/component/list-transition.ts
@@ -15,7 +15,7 @@ export interface ListTransitionProps<T> {
  * Props are accessed as getters (not destructured) so the compiler-generated
  * reactive getters are tracked by the underlying domEffect.
  */
-export function ListTransition<T>(props: ListTransitionProps<T>): Node {
+export function ListTransition<T>(props: ListTransitionProps<T>): DocumentFragment {
   const startMarker = document.createComment('lt-start');
   const endMarker = document.createComment('lt-end');
   const fragment = document.createDocumentFragment();

--- a/packages/ui/src/component/presence.ts
+++ b/packages/ui/src/component/presence.ts
@@ -15,7 +15,7 @@ export interface PresenceProps {
  * Props are accessed as getters (not destructured) so the compiler-generated
  * reactive getters are tracked by domEffect.
  */
-export function Presence(props: PresenceProps): Node {
+export function Presence(props: PresenceProps): DocumentFragment {
   const anchor = document.createComment('presence');
   let currentNode: HTMLElement | null = null;
   let exitingNode: HTMLElement | null = null;

